patches/tModLoader/Terraria/ModLoader/Default/TEModdedPylon.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System.Linq;
+using Terraria.DataStructures;
+using Terraria.ID;
+
+namespace Terraria.ModLoader.Default
+{
+	/// <summary>
+	/// This is a TML provided Tile Entity that acts extremely similar to vanilla's pylon TEs. If you plan
+	/// to make a pylon tile that closely resembles (or is identical to) vanilla's pylon TEs, you should use
+	/// this.
+	/// </summary>
+	public class TEModdedPylon : ModTileEntity, IPylonTileEntity
+	{
+
+		public override void NetPlaceEntityAttempt(int x, int y) {
+			if (!GetModPylonFromCoords(x, y, out ModPylon pylon)) {
+				RejectPlacementFromNet(x, y);
+				return;
+			}
+
+			int ID = Place(x, y, pylon.Mod);
+			NetMessage.SendData(MessageID.TileEntitySharing, number: ID, number2: x, number3: y);
+		}
+
+		public bool TryGetModPylon(out ModPylon modPylon) => GetModPylonFromCoords(Position.X, Position.Y, out modPylon);
+
+		private static void RejectPlacementFromNet(int x, int y) {
+			WorldGen.KillTile(x, y);
+			if (Main.netMode == NetmodeID.Server) {
+				NetMessage.SendData(MessageID.TileManipulation, number2: x, number3: y);
+			}
+		}
+
+		public int Place(int x, int y, Mod mod) {
+			int newID = Place(x, y);
+
+			(ByID[newID] as TEModdedPylon).Mod = mod;
+
+			Main.PylonSystem.RequestImmediateUpdate();
+			return newID;
+		}
+
+		private new int Place(int x, int y) {
+			TEModdedPylon moddedPylon = new TEModdedPylon();
+			moddedPylon.Position = new Point16(x, y);
+			moddedPylon.ID = AssignNewID();
+			moddedPylon.type = (byte)Type;
+
+			lock (EntityCreationLock) {
+				ByID[moddedPylon.ID] = moddedPylon;
+				ByPosition[moddedPylon.Position] = moddedPylon;
+			}
+
+			return moddedPylon.ID;
+		}
```

Comment:
This looks duplicated from `ModTileEntity.Place`, but without the benefit of cloning the `Mod` into it... If you just use that one then you shouldn't need to set `Mod` above